src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt

```diff
@@ -65,14 +65,25 @@ object PlayerListener : Listener {
             }
 
             transaction {
-                val playerCount = Players.select { bonfireUUID eq bonfire.uuid }.count()
-                if (playerCount >= BonfireConfig.data.maxPlayerCount) {
-                    return@transaction player.error("This bonfire is full!")
-                }else{
-                    player.setRespawnLocation(bonfire.uuid)
+                val playerBonfireUUID = Players
+                    .select { Players.playerUUID eq player.uniqueId }
+                    .firstOrNull()?.get(bonfireUUID) ?: return@transaction
```

Comment:
if a player has not set their respawn at a bonfire before, there will be no entry of them in the DB, meaning that `firstOrNull()` returns null. We want to set the player respawn in this case, but returning from the transaction doesn't do that. 

You can perhaps change this code to only get the first player row with `.firstOrNull()` without getting the bonfire UUID after, and then changing the if statements below to check `if (playerRow == null or playerRow.get(bonfireUUID) != bonfire.uuid) {set respawn} else {unset respawn}`

Reply:
should be fixed